src/main/java/ru/ayaz/UserSocketHandler.java

```diff
@@ -0,0 +1,129 @@
+package ru.ayaz;
+
+import ru.ayaz.ru.ayaz.exceptions.InvalidNicknameException;
+import ru.ayaz.ru.ayaz.exceptions.InvalidUserCommandException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+
+public class UserSocketHandler implements Runnable {
+
+    private ChatRoom room;
+    private PrintWriter writer;
+    private BufferedReader reader;
+    private User user;
+
+    UserSocketHandler(Socket socket, ChatRoom messageDistributor) throws IOException, InvalidNicknameException {
+        this.writer = new PrintWriter(socket.getOutputStream());
+        this.reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+        this.room = messageDistributor;
+        this.user = registerUser();
+    }
+
+    public User getUser() {
```

Comment:
&#x1F34E; **Issue**
This functionality is untested, although it can be easily tested.